docs/source/results/parallel.rst

```diff
@@ -0,0 +1,234 @@
+    $ wget https://raw.githubusercontent.com/bccp/nbodykit/master/nersc/example.py
+
+We can execute the example script with 4 MPI cores in parallel using:
+
+.. code:: bash
+
+    # run with 4 MPI workers
+    $ mpiexec -n 4 python example.py
+
+This example script generates a set of simulated objects and computes the
+power spectrum via the :class:`~nbodykit.algorithms.fftpower.FFTPower`
+algorithm, saving the result to a JSON file ``nbkit_example_power.json``.
+
+The MPI calling sequence typically differs for supercomputing environments
+and depends on the task scheduler being used. For example, when using the
+`Slurm manager <https://slurm.schedmd.com>`_ (as on NERSC), the equivalent
+command to execute the example script is:
+
+.. code:: bash
+
+    $ srun -n 4 python example.py
+
+On a managed computing facility, it is also usually necessary to include
+directives that reserves the computing resource used for running the script.
+Often, you can refer to the 'Submitting a Job' section of the user guide of
+the facility, e.g., `NERSC Cori`_, `NERSC Edison`_, and the `NCSA`_.
+However, be aware these user guides are not always accurate,
+and it is always better to check with someone who uses these facilities first.
+
+
+.. _NERSC Cori: http://www.nersc.gov/users/computational-systems/cori/running-jobs/
+.. _NERSC Edison: http://www.nersc.gov/users/computational-systems/edison/running-jobs/
+.. _NCSA: https://bluewaters.ncsa.illinois.edu/getting-started/#Running
+
+A Primer on MPI Communication
+-----------------------------
+
+MPI stands for Message Passage Interface, and unsurprisingly, one of its
+key elements is the communication between processes running in parallel.
+The MPI standard allows processes running in parallel, which own their
+own memory, to exchange messages, thus allowing the independent results
+computed by individual processes to be combined into a single result.
+
+The MPI communicator object is responsible for managing the communication
+of data and messages between parallel processes. In nbodykit, we manage the
+current MPI communicator using the :class:`nbodykit.CurrentMPIComm` class.
+By default, all nbodykit objects, i.e., catalog and mesh objects, use
+the communicator defined by this class to exchange results. For
+catalog and mesh objects, the communicator is stored as the :attr:`comm`
+attribute.
+
+Users can access the current communicator by calling the
+:func:`~nbodykit.CurrentMPIComm.get` function of the
+:class:`~nbodykit.CurrentMPIComm` object. For example,
+
+.. code:: python
+
+    from nbodykit import CurrentMPIComm
+    comm = CurrentMPIComm.get()
+
+The communicator object carries a ``rank`` attribute, which provides
+a unique numbering of the available processes controlled
+by the communicator. For example, this attribute is often used to filter out
+and control the behavior of a "root" rank:
+
+.. code:: python
+
+    if comm.rank == 0:
+      print("I am the root rank!")
```

Comment:
Often used to reduce the amount of messages printed to the terminal. e.g.

if comm.rank == 0:
   print("I am Groot.")

We get one. without the condition will get as many as `comm.size` messages.

Reply:
😆